lib/components/form/call-taker/date-time-picker.tsx

```diff
@@ -0,0 +1,235 @@
+ * - the time input will interpret various time formats so that
+ *   users can quickly type a shorthand value (5p) and have that be parsed into
+ *   the correct OTP format.
+ * - when a user changes the date or time, "leave now" (if selected) will
+ *   automatically switch to "depart at".
+
+ * @type {Object}
+ */
+
+const DateTimeOptions = ({
+  date: initialDate,
+  homeTimezone,
+  onKeyDown,
+  setQueryParam,
+  time: initialTime,
+  timeFormat
+}: Props) => {
+  const [departArrive, setDepartArrive] = useState(
+    initialDate || initialTime ? 'DEPART' : 'NOW'
+  )
+  const [date, setDate] = useState<string | undefined>(initialDate)
+  const [time, setTime] = useState<string | undefined>(initialTime)
+
+  const intl = useIntl()
+
+  /**
+   * Parse a time input expressed in the agency time zone.
+   * @returns A date if the parsing succeeded, or null.
+   */
+  const parseInputAsTime = (
+    timeInput: string = getCurrentTime(homeTimezone),
+    date: string = getCurrentDate(homeTimezone)
+  ) => {
+    if (!timeInput) timeInput = getCurrentTime(homeTimezone)
+
+    // Match one of the supported time formats
+    const matchedTimeFormat = SUPPORTED_TIME_FORMATS.find((timeFormat) =>
+      isMatch(timeInput, timeFormat)
+    )
+    if (matchedTimeFormat) {
+      const resolvedDateTime = format(
+        parse(timeInput, matchedTimeFormat, new Date()),
+        'HH:mm:ss'
+      )
+      return toDate(`${date}T${resolvedDateTime}`)
+    }
+    return ''
+  }
+
+  const dateTime = parseInputAsTime(time, date)
+
+  // Handler for setting the query parameters
+  useEffect(() => {
+    if (safeFormat(dateTime, OTP_API_DATE_FORMAT, {}) !== '' && setQueryParam) {
+      setQueryParam({
+        date: safeFormat(dateTime, OTP_API_DATE_FORMAT, {
+          timeZone: homeTimezone
+        }),
+        departArrive,
+        time: safeFormat(dateTime, OTP_API_TIME_FORMAT, {
+          timeZone: homeTimezone
+        })
+      })
+    }
+  }, [dateTime, departArrive, homeTimezone, setQueryParam])
+
+  // Handler for updating the time and date fields when NOW is selected
+  useEffect(() => {
+    if (departArrive === 'NOW') {
+      setTime(getCurrentTime(homeTimezone))
+      setDate(getCurrentDate(homeTimezone))
+    }
+  }, [departArrive, setTime, setDate, homeTimezone])
+
+  const unsetNow = () => {
+    if (departArrive === 'NOW') setDepartArrive('DEPART')
+  }
+
+  return (
+    <>
+      <select
+        onBlur={(e) => setDepartArrive(e.target.value)}
+        onChange={(e) => setDepartArrive(e.target.value)}
+        onKeyDown={onKeyDown}
+        value={departArrive}
+      >
+        {getDepartureOptions(intl).map(({ text, value }) => (
+          <option key={value} value={value}>
+            {text}
+          </option>
+        ))}
+      </select>
+      <OverlayTrigger
+        overlay={
+          <Tooltip id="time-tooltip">
+            {safeFormat(dateTime, timeFormat, {
+              timeZone: homeTimezone
+            })}
```

Comment:
Would it be helpful to have a little error msg get rendered here instead of an empty tooltip when the time format doesn't work? Something like 
   ```
        <Tooltip id="time-tooltip">
            {dateTime ? safeFormat(dateTime, timeFormat, {
              timeZone: homeTimezone
            }) : "Invalid Time"}
        </Tooltip>
```
I'll defer to you, I think it works okay as is!

Reply:
This is a huge ux improvement thanks so much! Implemented